source/javascripts/components/unified-editor/StepConfigDrawer/StepConfigDrawer.context.tsx

```diff
@@ -25,7 +28,26 @@ const StepConfigDrawerProvider = ({ children, workflowId, stepIndex }: PropsWith
         : initialState,
     [workflowId, stepIndex, result],
   );
-  return <Context.Provider value={value}>{children}</Context.Provider>;
+
+  useEffect(() => {
+    form.reset({
+      configuration: {
+        is_always_run: result.data?.mergedValues?.is_always_run ?? false,
+        is_skippable: result.data?.mergedValues?.is_skippable ?? false,
+        run_if: result.data?.mergedValues?.run_if ?? '',
+      },
+      properties: {
+        name: result.data?.resolvedInfo?.title ?? '',
+        version: result.data?.resolvedInfo?.normalizedVersion ?? '',
+      },
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [result.data?.resolvedInfo?.title, result.data?.resolvedInfo?.normalizedVersion]);
```

Comment:
I think it's better if you reset the form when the `workflowId` or the `stepIndex` was changed.